frontend/modules/viewer/viewer.redux.ts

```diff
@@ -22,6 +22,7 @@ export const { Types: ViewerTypes, Creators: ViewerActions } = createActions({
 	updateSettings: ['username', 'settings'],
 	updateSettingsSuccess: ['settings'],
 	fetchSettings: [],
+	updatePins: ['prev', 'curr'],
```

Comment:
@sanmont3drepo  Does this make sense here? With the latest changes you've done with the pin fix

Reply:
Yes, it doesn't align with they idea behind how pins are working @m1chalkubiak . Thats why I got rid from the "updatepins" bit from sagas.

The general concept is this:
- in the state (redux) there is only the information of the issues and risks and positions (which will later become the pins.)
- in the selector the information from the issues and risks/ filtering of them and show/hide pins state is being consumed and returns which pins should be visible at that moment.
- this information is being used in the viewer react component to actually show and hide the pins, acting like any react component (representing visually the information from the state)
- the previous/current pins is part of the mechanism in the visual representation layer, it has nothing to do with state. Its actually how react components work (difference between the prev and current state and managing the dom and how that it would be represented) 

As a minor exercise to see clearly that curr and prev is from  the visual representation layer and has nothing to do with state, you could change it to instead of just checking the difference between the previous and current state and adding/removing the pins, you could just wipeout all the pins and add all the pins from the current  state.